src/librustc_mir/const_eval.rs

```diff
@@ -45,9 +45,15 @@ fn mk_eval_cx<'mir, 'tcx>(
     tcx: TyCtxt<'tcx>,
```

Comment:
Just opening this tot rack fixing the PR title -- what this really checks is that computation and validation doesn't *read from* statics, which is not the same as checking the final value.